tests/Unit/Trace/SpanProcessor/ResponseTimeSpanProcessorTest.php

```diff
@@ -0,0 +1,274 @@
+        $spanData = Mockery::mock(SpanDataInterface::class);
+        $spanData->expects('getStatus')->andReturn(StatusData::create(StatusCode::STATUS_OK, 'ok'));
+        $this->readableSpan->expects('toSpanData')->andReturn($spanData);
+        $this->readableSpan->expects('getKind')->andReturn(SpanKind::KIND_SERVER);
+        $this->readableSpan->expects('getAttribute')->with(TransactionNameSpanProcessor::TRANSACTION_NAME_ATTRIBUTE)->andReturn('transaction');
+        $this->readableSpan->expects('getAttribute')->with(TraceAttributes::HTTP_METHOD)->andReturn('GET');
+        $this->readableSpan->expects('getAttribute')->with(TraceAttributes::HTTP_STATUS_CODE)->andReturn(200);
+        $this->readableSpan->expects('getAttribute')->with(TraceAttributes::HTTP_REQUEST_METHOD)->andReturnNull();
+        $this->readableSpan->expects('getAttribute')->with(TraceAttributes::HTTP_RESPONSE_STATUS_CODE)->andReturnNull();
+        $this->responseTimeSpanProcessor->onEnd($this->readableSpan);
+        $this->reader->collect();
+        $metrics = $this->exporter->Collect();
+        // Check histogram
+        $this->assertCount(1, $metrics);
+        foreach ($metrics as $metric) {
+            $this->assertEquals('trace.service.response_time', $metric->name);
+            $this->assertEquals('ms', $metric->unit);
+            if ($metric->data instanceof Histogram) {
+                $this->assertEquals('Delta', $metric->data->temporality);
+                $this->assertCount(1, $metric->data->dataPoints);
+                foreach ($metric->data->dataPoints as $dataPoint) {
+                    $this->assertEquals(123.456789, $dataPoint->sum);
+                    $attributes = [];
+                    foreach ($dataPoint->attributes as $key => $value) {
+                        $attributes[$key] = $value;
+                    }
+                    $this->assertFalse($attributes['sw.is_error'] ?? true);
+                    $this->assertEquals('transaction', $attributes[TransactionNameSpanProcessor::TRANSACTION_NAME_ATTRIBUTE] ?? '');
+                    $this->assertEquals('GET', $attributes[TraceAttributes::HTTP_METHOD] ?? '');
+                    $this->assertEquals(200, $attributes[TraceAttributes::HTTP_STATUS_CODE] ?? '');
+                    $this->assertArrayNotHasKey(TraceAttributes::HTTP_REQUEST_METHOD, $attributes);
+                    $this->assertArrayNotHasKey(TraceAttributes::HTTP_RESPONSE_STATUS_CODE, $attributes);
+                }
+            }
+        }
+    }
+    public function test_local_parent(): void
+    {
+        $generator = new RandomIdGenerator();
+        $validLocalParentContext = SpanContext::create(
+            $generator->generateTraceId(),
+            $generator->generateSpanId(),
+            TraceFlags::SAMPLED
+        );
+        $this->readableSpan->expects('getParentContext')->andReturn($validLocalParentContext);
+        $this->responseTimeSpanProcessor->onEnd($this->readableSpan);
```

Comment:
If current span has a valid local parent, then I don't think it should get `response_time` attribute. Current span does get `response_time` if its valid parent is remote, like the next test.

Reply:
The histogram is created when ResponseTimeSpanProcessor is constructed. So the response_time attribute exists, but the dataPoints count is 0.

Reply:
Ahh ok. The attribute exists in collected metrics, but not the current span. Is that because of the previous test? Or are empty metrics exported by upstream? Or something else?

Reply:
I was wondering if previous tests in this file like `test_span_ok_depreciated_fields` or above ran before this test so that `$metrics` would have `trace.service.response_time` populated in a previous batch of exported metrics, leaving record of it. But I think I'm just confusing myself. Instead, `trace.service.response_time` histogram is created at app start up, then exports no data points for this span like you've said. So I think this is fine, thanks for clarifying.